### PR TITLE
feat(codeowners): Rm CodeOwners CTA hooks

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
Required for https://github.com/getsentry/getsentry/pull/7327

Updates the CTA in Issue Details page with more complex states. I also expanded the audience to include free plan users to see if we can get some conversions from this feature.

Developer Plan
codeowners cta - issue details - developer

Team Plan
codeowners cta - issue details - team

Business Plan / Also the default for self hosted
codeowners cta - issue details - setup now

Business Plan - Request Setup
codeowners cta - issue details - request setup

Removes the following CTA:
codeowners banner - on business

### CODE REVIEW

**Review by ceorourke — APPROVED** [review]

**Comment by evanpurkhiser** [comment]
I think this is unrelated, but why are the button sizes different in these?

**Comment by NisanthanNanthakumar** [comment]
> I think this is unrelated, but why are the button sizes different in these?

@evanpurkhiser this is fixed with https://github.com/getsentry/getsentry/pull/7327/commits/728b928608b66fff48a161bb45ec0a73358f9569

**Comment by ceorourke** [comment]
I'm worried my initial approval before it got more complex is throwing people off of reviewing so I'm gonna pick on @leeandher 's front endier eyes to take a peek.

File: static/app/views/settings/project/projectOwnership/addCodeOwnerModal.tsx [hunk 1]
```diff
@@ -189,37 +207,50 @@ class AddCodeOwnerModal extends Component<Props, State> {
...
-                    type="button"
-                    to={`${baseUrl}/${integration.provider.key}/${integration.id}/?tab=codeMappings&referrer=add-codeowners`}
-                  >
-                    {getIntegrationIcon(integration.provider.key)}
-                    <IntegrationName>{integration.name}</IntegrationName>
+          {!codeMappings.length ? (
+            !integrations.length ? (
+              <Fragment>
+                <div>
+                  {t('Install a GitHub or GitLab integration to use this feature.')}
```
Comment (leeandher) [inline static/app/views/settings/project/projectOwnership/addCodeOwnerModal.tsx:224]: Should we change this to 'source control integration'? I know we only support GitHub and GitLab right now, but things change down the line we'll have to remember to update this. Just curious of your thoughts

File: static/app/views/settings/project/projectOwnership/addCodeOwnerModal.tsx [hunk 2]
```diff
@@ -189,37 +207,50 @@ class AddCodeOwnerModal extends Component<Props, State> {
...
+                </div>
+                <Container style={{paddingTop: space(2)}}>
+                  <Button type="button" priority="primary" size="small" to={baseUrl}>
+                    Setup Integrations
```
Comment (leeandher) [inline static/app/views/settings/project/projectOwnership/addCodeOwnerModal.tsx:228]: Should it be `Setup Integration` (not plural)? They only need one, right?

File: static/app/components/group/suggestedOwners/ownershipRules.tsx [hunk 3]
```diff
@@ -27,6 +27,26 @@ type Props = {
   project: Project;
 };
 
+const CodeOwnersCTA = HookOrDefault({
```
Comment (leeandher) [inline static/app/components/group/suggestedOwners/ownershipRules.tsx:30]: I didn't know we had this pattern, this is kinda cool :+1:

**Review by leeandher — COMMENTED** [review]
@NisanthanNanthakumar If we're removing these new badges, should we also remove the badges that appear on the  issue owners page's section headers?

The only thing that is new is auto-sync, so I think the sidebar badge and auto-sync badge can stay.

image

File: static/app/views/settings/project/projectOwnership/addCodeOwnerModal.tsx (hunk 1 above)
Comment (NisanthanNanthakumar) [reply static/app/views/settings/project/projectOwnership/addCodeOwnerModal.tsx:224]: I think we should keep it specific bc I don't see CODEOWNERS support on the roadmap for Bitbucket/Azure Devops. We can update the copy when/if CodeOwners needs to expanded to other source control integrations.

File: static/app/views/settings/project/projectOwnership/addCodeOwnerModal.tsx (hunk 2 above)
Comment (NisanthanNanthakumar) [reply static/app/views/settings/project/projectOwnership/addCodeOwnerModal.tsx:228]: Yes, but I want them to install more integrations! haha I'll change this!

**Comment by NisanthanNanthakumar** [comment]
> If we're removing these new badges, should we also remove the badges that appear on the issue owners page's section headers?

The only thing that is new is auto-sync, so I think the sidebar badge and auto-sync badge can stay.

@leeandher I removed the new badge on the sidebar and the panel section headers. I got feedback from design to remove it from the sidebar. But we'll keep the badge for auto-sync for a little longer!

**Review by leeandher — APPROVED** [review]
Looks good! @NisanthanNanthakumar, agreed about the badges you chose to keep btw, auto-sync deserves the attention lol

File: static/app/components/group/suggestedOwners/ownershipRules.tsx [hunk 4]
```diff
@@ -27,6 +27,26 @@ type Props = {
...
+  hookName: 'component:codeowners-cta',
+  defaultComponent: ({organization, project}) => (
+    <SetupButton
+      size="xsmall"
+      priority="primary"
+      to={`/settings/${organization.slug}/projects/${project.slug}/ownership/`}
+      onClick={() =>
+        trackIntegrationAnalytics('integrations.code_owners_cta_setup_clicked', {
```
Comment (scefali) [inline static/app/components/group/suggestedOwners/ownershipRules.tsx:38]: @NisanthanNanthakumar you don't need analytics on the default component tbh, this is only going to be used for self-hosted and development

**Pull request merged**